### ISSUE TITLE
comparison to svtyper

### ISSUE THREAD

**Issue by brentp** [Illumina/paragraph#20]
I use svtyper in [smoove](https://github.com/brentp/smoove). After reading your paper, I thought I might replace svtyper with paragraph.
I did a separate evaluation using the GiaB truthset from [here](ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/AshkenazimTrio/analysis/NIST_SVs_Integration_v0.6/)
and using truvari.
I evaluated only on deletions > 300 bases. 

When genotyping this large-DEL truthset. I get 81% recall from paragraph and 91% with svtyper.
I realize that you used a different call-set and not limiting to Tier 1 regions, but I am surprised the results are so different. I am wondering if you have any insight on this.

 I used paragraph via the docker image (as updated with my pending pull-request) and the code below:

```
wget ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/AshkenazimTrio/analysis/NIST_SVs_Integration_v0.6/HG002_SVs_Tier1_v0.6.vcf.gz
bcftools view -f "PASS,." -O z -o HG002_SVs_Tier1_v0.6.DEL.vcf.gz -i 'SVTYPE == "DEL" & SVLEN < -40' HG002_SVs_Tier1_v0.6.vcf.gz
# paragraph complains about reference so manually change:
zcat HG002_SVs_Tier1_v0.6.DEL.vcf.gz | awk 'BEGIN{FS=OFS="\t"} ($0 ~ /^#/) { print } ($0 !~ /^#/ ) { $4="N"; $5 = "<DEL>"; print }' | bgzip -c > tmp
mv tmp HG002_SVs_Tier1_v0.6.DEL.vcf.gz
docker run -v $(pwd):/pwd -v /data/human:/data/human 5a75c4ae6ebc -m /pwd/manifest.txt -r /data/human/g1k_v37_decoy.fa --threads 4 -o /pwd/ -i /pwd/HG002_SVs_Tier1_v0.6.DEL.vcf.gz
wget ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/AshkenazimTrio/analysis/NIST_SVs_Integration_v0.6/HG002_SVs_Tier1_v0.6.bed

truth_del=HG002_SVs_Tier1_v0.6.DEL.vcf.gz
ODIR=evaluate/
sizemax=15000000
sizemin=300
bed=HG002_SVs_Tier1_v0.6.bed

set -euo pipefail

rm -r $ODIR
tabix -f  genotypes.vcf.gz
  
python ~/src/truvari/truvari.py --sizemax $sizemax -s $sizemin -S $((sizemin - 30)) -b $truth_del -c genotypes.vcf.gz -o $ODIR/ --passonly --pctsim=0 -r 20 --giabreport -f /data/human/g1k_v37_decoy.fa --no-ref --includebed $bed -O 0.6
cat $ODIR/summary.txt

zcat $truth_del | ./add_ci 

svtyper -B /data/human/hg002.cram -T /data/human/g1k_v37_decoy.fa \
	-i with-ci.vcf \
	--max_ci_dist 0 \
	-o svtyper.genotyped.vcf

perl -pi -e 's/""/"/' svtyper.genotyped.vcf
bgzip svtyper.genotyped.vcf
tabix -f svtyper.genotyped.vcf.gz

ODIR=evaluate-svtyper/

rm -r $ODIR
python ~/src/truvari/truvari.py --sizemax $sizemax -s $sizemin -S $((sizemin - 30)) -b $truth_del -c svtyper.genotyped.vcf.gz -o $ODIR/ --passonly --pctsim=0  -r 20 --giabreport -f /data/human/g1k_v37_decoy.fa --no-ref --includebed $bed -O 0.6
cat $ODIR/summary.txt
```

**Comment by brentp** [comment]
and here is the source of add_ci to get around svtyper requiring CIPOS and CIEND and a header parsing bug in svtyper:
```Nim
import hts/vcf

var ivcf:VCF
var ovcf:VCF

if not open(ivcf, "/dev/stdin"):
  quit "bad"

doAssert ivcf.header.add_info("CIPOS", "2", "Integer", "ci") == Status.OK
doAssert ivcf.header.add_info("CIEND", "2", "Integer", "ci") == Status.OK
doAssert ivcf.header.add_info("CIPOS95", "2", "Integer", "ci") == Status.OK
doAssert ivcf.header.add_info("CIEND95", "2", "Integer", "ci") == Status.OK
doAssert ivcf.header.remove_info("MultiTechExact") == Status.OK
doAssert ivcf.header.remove_info("MultiTech") == Status.OK
doAssert ivcf.header.remove_info("DistPASSHG2gt49Minlt1000") == Status.OK


if not open(ovcf, "with-ci.vcf", mode="w"):
  quit "bad"


ovcf.copy_header(ivcf.header)
doAssert ovcf.write_header()

var ci = @[-2'i32, 2]
for v in ivcf:
  doAssert v.info.set("CIPOS", ci) == Status.OK
  doAssert v.info.set("CIEND", ci) == Status.OK
  doAssert v.info.set("CIPOS95", ci) == Status.OK
  doAssert v.info.set("CIEND95", ci) == Status.OK
  discard v.info.delete("MultiTech")
  discard v.info.delete("MultiTechExact")
  discard v.info.delete("DistPASSHG2gt49Minlt1000")
  doAssert ovcf.write_variant(v)

ovcf.close()
```

**Comment by traxexx** [comment]
Hi Brent,

Thanks for looking into this!

For the recall of Paragraph, we got similar results to yours on NIST truth set
v0.6 (see supplementary materials). NIST truth set is not guaranteed
to be breakpoint accurate, so some Paragraph FNs are likely due to inaccurate
breakpoints.

We also observed that svtyper has a good recall for >300bp deletions (Fig 2a)
but for smaller deletions its performance appears to drop sharply. Since a large
fraction of deletions is smaller than 300bp (Fig 2b) we get a much lower
estimate for the overall recall (Table 1).

I'd say the test set matters too. NIST tier1 contains only confident regions of
the genome. And deletion size is important when making such comparisons.

And we indeed added CIPOS & CIEND when evaluating svtyper.

**Comment by brentp** [comment]
hi, thanks for the response, is a VCF of the LRGT set available?
I don't see svtyper deletions on GiaB set in the supplement, only Manta and Paragraph in table S2.

on lumpy output (where svtyper genotypes result in ~90% precision and 84% recall), paragraphs gives only 20% recall and 98% precision.

I will likely leave this for now, feel free to close the issue. It would have saved me some time to know that paragraph currently won't work with lumpy output (due to lack of precision in break-points?)

**Comment by fritzsedlazeck** [comment]
Hey @brentp 
thanks for looking at this. Always interesting to see how things run in other peoples hands. 
Our calls are here: https://github.com/Illumina/paragraph/blob/master/data/download-instructions.txt 

Would be interesting to know why the calls from Lumpy are showing reduced performance. 
Thanks
Fritz

**Comment by brentp** [comment]
hi Fritz, thanks for the reply. If you make the cram/bam that you used available, I'll be glad to retry the evaluation on those variants+alignments.

**Comment by fritzsedlazeck** [comment]
you mean the Pacbio data?

**Comment by brentp** [comment]
I mean the illumina hg002 ~35X sample.

**Comment by traxexx** [comment]
@brentp Yes, HG002 Long-read ground truth is available in data/ directory. min event length = 30bp. del+ins+inv+dup. Note that in the paper we only used 50~10kbp del & ins events for evaluation.

And in Table S2 we only tested Manta & Paragraph. We didn't test everything since such comparison has been done on LRGT.

It's interesting to see such a different recall on lumpy calls. We never tested our method on lumpy calls before. For now, I guess it's mostly because of breakpoints, as the PE method is unlikely to achieve base-pair accuracy. But we're going to double check.
